crates/dbsp/src/ir.rs

```diff
@@ -0,0 +1,179 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use feldera_ir::{LirCircuit, LirEdge, LirNode, LirStreamId, MirNodeId};
+
+use crate::{
+    circuit::{circuit_builder::CircuitBase, GlobalNodeId, NodeId},
+    Circuit, Stream,
+};
+
+/// Node label indicating that this node implements the output of an
+/// MIR operator.
+pub const LABEL_MIR_NODE_ID: &str = "mir_node";
+
+impl<C, T> Stream<C, T>
+where
+    C: Circuit,
+{
+    /// Sets `id` as both the MIR node id and persistent id of the
+    /// source node of the stream.
+    pub fn set_mir_node_id(&self, id: &str) {
```

Comment:
I find that "persistent" is important, should it be in the name of the function?

Reply:
`ser_persistent_mir_id`?